### ISSUE TITLE
Create an `asRepositoryName()` helper

### ISSUE THREAD

**Issue by angeloashmore** [prismicio/prismic-helpers#42]
### Is your feature request related to a problem? Please describe.

When working with Prismic and its API, you may need full repository API endpoint (e.g. `https://qwerty.cdn.prismic.io/api/v2`) or just the repository name (e.g. `qwerty`).

`@prismicio/client` provides a function to create an API endpoint: `prismic.getEndpoint("qwerty")`

There is no option to go in the reverse direction, however. If you have an API endpoint, you can only get the repository name by using a regex or creating a `URL` instance. This is verbose and could be error prone.

### Describe the solution you'd like

A helper to extract a repository name from a standard Prismic Rest API V2 endpoint could be helpful.

Such a helper could look like the following (the function name is up for debate):

```ts
import * as prismicH from "@prismicio/helpers"

const endpoint = "https://qwerty.cdn.prismic.io/api/v2";
const repositoryName = prismicH.asRepositoryName(endpoint)
// => "qwerty"
```

Note that this does not support any non-standard Prismic Rest API V2 endpoints. If a user has a custom proxy endpoint, for example, this function cannot help.

It does not perform a network request to resolve the repository name. It simply does string manipulation to extract that part it needs.

The above would be equivalent to the following:

```ts
const endpoint = "https://qwerty.cdn.prismic.io/api/v2";
const repositoryName = new URL(endpoint).hostname.split('.')[0]
```

### Describe alternatives you've considered

The `new URL()` method shown above is the alternative. It is noisy and quite imperative.

### Additional context

This could be useful specifically for Slice Machine projects. Slice Machine's config file, `sm.json`, includes an `apiEndpoint` property (at least it does at the time of this issue). This proposed helper function would allow a user to import `sm.json` into their project to reuse the API endpoint and the repository name.

**Comment by lihbr** [comment]
I think this should be part of `@prismicio/client`, not `@prismicio/helpers` (which targets more the templating of Prismic data)

Along the `getEndpoint()` function, we could add `getRepositoryID()`, `isEndpoint()` and `isRepositoryID()`.

For consistency maybe we could rename `getEndpoint()` to `getRepositoryEndpoint()` (same for `isEndpoint()` then).

I also think "repository ID" is more accurate than "repository name" which is a value that also exists in the UI, but no strong opinion on that 🤔

Whole API could look like this:

```typescript
getRepositoryEndpoint(repositoryID: string) => string;
getRepositoryID(repositoryEndpoint: string) => string;

isRepositoryEndpoint(repositoryID: string) => boolean;
isRepositoryID(repositoryEndpoint: string) => boolean;

/** @deprecated */
getEndpoint = getRepositoryEndpoint;
```

And for simplicity I'd be happy for `createClient()` to support both a repository endpoint or a repository ID, allowing us to just skip the `getEndpoint()` part in the education process? (I know there are arguments against it, but I just keep doing it on every kit I work on 😢)

**Comment by angeloashmore** [comment]
All of that makes sense to me!

Re: "repository ID" vs "repository name" - Although "ID" is technically more correct, "name" is used throughout documentation and the UI. Using "ID" here might be confusing compared to keeping it consistent.

I implemented the suggestions in `@prismicio/client` here: https://github.com/prismicio/prismic-client/pull/222

I also implemented the auto endpoint functionality here: https://github.com/prismicio/prismic-client/pull/223 (depends on the previous PR)

**Comment by angeloashmore** [comment]
This is shipped in `v6.3.0` as `getRepositoryName()` alongside these endpoint helpers:

- **`getRepositoryEndpoint()`**: Rename of `getEndpoint()`
- **`getRepositoryName()`**: Extracts the repository name from a standard Prismic Rest API V2 endpoint
- **`isRepositoryEndpoint()`**: Determines if an input is a valid repository endpoint
- **`isRepositoryName()`**: Determines if an input is a valid repository name
